### PR TITLE
NIST parser new HDF output

### PR DESCRIPTION
**Pull request by epassaro**
- [x] Ionization energies
- [x] Atomic weights
- [x] Try to inherit from `BaseParser` class
- [x] Notebook for documentation
https://nbviewer.jupyter.org/github/epassaro/carsus/blob/nist-hdf/docs/io/nist/nist_weights_comp.ipynb?flush_cache=true
- [x] Write docstrings
- [x] PEP 8
- [ ] ready to merge

### CODE REVIEW

File: carsus/io/nist/ionization.py [hunk 1]
```diff
@@ -290,3 +291,35 @@ def ingest(self, ionization_energies=True, ground_levels=True):
...
+
+class NISTIonizationEnergies(BaseParser):
+    """
+    Attributes
+    ----------
+    base : pandas.Series
+
+    Methods
+    -------
+    to_hdf(fname)
```
Comment (wkerzendorf) [inline carsus/io/nist/ionization.py]: colon missing?

Comment (epassaro) [reply carsus/io/nist/ionization.py]: I'm using the same docstring format used in 'old' classes such as `NISTIonizationEnergiesIngester`. 

In the *numpydoc* guide the methods section says: `Document these as you would any other function`, so I assume this is not the appropiate format. 

I'll fix this.

Comment (epassaro) [reply carsus/io/nist/ionization.py]: Forgot that. This is an example provided in the *numpydocs* guide:

```
class Photo(ndarray):
    """
    Array with associated photographic information.

    ...

    Attributes
    ----------
    exposure : float
        Exposure in seconds.

    Methods
    -------
    colorspace(c='rgb')
        Represent the photo in the given colorspace.
    gamma(n=1.0)
        Change the photo's gamma exposure.

    """
```

Comment (epassaro) [reply carsus/io/nist/ionization.py]: Ok, done!

File: carsus/io/nist/ionization.py [hunk 2]
```diff
@@ -290,3 +291,59 @@ def ingest(self, ionization_energies=True, ground_levels=True):
...
+    def __init__(self, spectra):
+        input_data = download_ionization_energies(spectra)
+        self.parser = NISTIonizationEnergiesParser(input_data)
+        self._prepare_data()
+
+    def _prepare_data(self):
+        ionization_data = pd.DataFrame()
+        ionization_data['atomic_number'] = self.parser.base['atomic_number']
+        ionization_data['ion_number'] = self.parser.base['ion_charge'] + 1
+        ionization_data['ionization_energy'] = self.parser.base['ionization_energy_str'].str.strip('[]()').astype(np.float64)
```
Comment (wkerzendorf) [inline carsus/io/nist/ionization.py]: PEP80 - more than 80 characters ;-)

File: carsus/io/nist/weightscomp.py [hunk 3]
```diff
@@ -183,3 +186,56 @@ def ingest(self, atomic_weights=True):
...
+        atomic_numbers = parse_selected_atoms(atoms)
+        atom_data_list = []
+
+        for atomic_number in atomic_numbers:
+            basic_atomic_data = pd.read_csv(basic_atomic_data_fname)
+            basic_atomic_data = basic_atomic_data.loc[atomic_number-1]
+
+            atom_masses = self.parser.prepare_atomic_dataframe()
+            atom_masses = atom_masses.drop(columns='atomic_weight_std_dev')
+            atom_masses = atom_masses.rename(columns={'atomic_weight_nominal_value': 'mass'})
```
Comment (wkerzendorf) [inline carsus/io/nist/weightscomp.py]: more than 80 char?

File: docs/io/nist/nist_weights_comp.ipynb [hunk 4]
```diff
@@ -1,10 +1,23 @@
 {
  "cells": [
+  {
```
Comment (wkerzendorf) [inline docs/io/nist/nist_weights_comp.ipynb]: Remove the sqlite stufff

Comment (epassaro) [reply docs/io/nist/nist_weights_comp.ipynb]: Done

File: carsus/io/nist/ionization.py (hunk 2 above)
Comment (epassaro) [reply carsus/io/nist/ionization.py]: Done

File: carsus/io/nist/weightscomp.py [hunk 5]
```diff
@@ -8,9 +8,12 @@
...
 from bs4 import BeautifulSoup
 from astropy import units as u
+from carsus.base import basic_atomic_data_fname
 from carsus.model import AtomWeight
-from carsus.io.base import BasePyparser, BaseIngester
+from carsus.util import parse_selected_atoms
+from carsus.io.base import BasePyparser, BaseIngester, BaseParser
 from carsus.io.util import to_nom_val_and_std_dev
+from carsus.util.helpers import ATOMIC_SYMBOLS_DATA
 from carsus.io.nist.weightscomp_grammar import isotope, COLUMNS, ATOM_NUM_COL, MASS_NUM_COL,\
```
Comment (marxwillia) [inline carsus/io/nist/weightscomp.py]: PEP8 line length.

Comment (epassaro) [reply carsus/io/nist/weightscomp.py]: I didn't touch any previous existing code, applied PEP8 standards to my changes only. There are a LOT of long lines!

File: carsus/io/nist/__init__.py [hunk 6]
```diff
@@ -1,2 +1,8 @@
-from .weightscomp import download_weightscomp, NISTWeightsCompPyparser, NISTWeightsCompIngester
-from .ionization import download_ionization_energies, NISTIonizationEnergiesParser, NISTIonizationEnergiesIngester
\ No newline at end of file
+from carsus.io.nist.weightscomp import (download_weightscomp, 
+                                   NISTWeightsCompPyparser,
```
Comment (marxwillia) [inline carsus/io/nist/__init__.py]: Make indentation consistent with above line.

**Pull request merged**